### PR TITLE
Simplify callbacks

### PR DESCRIPTION
**Pull request by tpluscode**
Prompted by Blake's [message on Gitter](https://gitter.im/rdfjs/Representation-Task-Force?at=60e3cbbe8a40b11728478815) I thought I would take a second look and apparently the current types are imperfect

By inlining the callback/iteratee functions, the correct type of `dataset` argument can be used.

### CODE REVIEW

File: .changeset/seven-shrimps-build.md [hunk 1]
```diff
@@ -0,0 +1,5 @@
+---
+"@rdfjs/types": minor
```
Comment (tpluscode) [inline .changeset/seven-shrimps-build.md:2]: Not sure whether this should be minor or patch. It's a significant change but maybe still just a bug fix?

**Comment by blake-regalia** [comment]
The problem seems to stem from the spec, could this have been a limitation of WebIDL? https://rdf.js.org/dataset-spec/#quadfilteriteratee-interface

**Comment by tpluscode** [comment]
Possibly. And apparently it got translated to TS verbatim

File: .changeset/seven-shrimps-build.md (hunk 1 above)
Comment (blake-regalia) [reply .changeset/seven-shrimps-build.md:2]: I'm ok with classifying it as minor

**Review by blake-regalia — APPROVED** [review]

**Pull request merged**